pkg/util/maps.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2020 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+)
+
+const (
+	keyPairSeparator = "="
+	pairSeparator    = ","
+)
+
+// MapContainsMap returns true only if source contains expected map
+func MapContainsMap(source, expected map[string]string) bool {
+	if len(source) == 0 && len(expected) == 0 {
+		return false
+	}
+	if len(source) > 0 && len(expected) == 0 {
+		return false
+	}
+	if len(source) == 0 && len(expected) > 0 {
+		return false
+	}
+	for k, v := range expected {
+		if source[k] != v {
+			return false
+		}
+	}
+	return true
+}
+
+// FromMapToString converts a map into a string format such as key1=value1,key2=value2
+func FromMapToString(labels map[string]string) string {
+	b := new(bytes.Buffer)
+	for k, v := range labels {
+		fmt.Fprintf(b, "%s=%s%s", k, v, pairSeparator)
+	}
+	return strings.TrimSuffix(b.String(), pairSeparator)
+}
+
+// FromStringsKeyPairToMap converts a string array in the key/pair format (key=value) to a map. Unconvertable strings will be skipped.
```

Comment:
If inconvertible strings are skipped, shouldn't we use continue in the loop?

Reply:
This code has been moved to this new file. I won't touch on this functions in this JIRA to not insert regressions since it's used by CLI, out of the scope to test it now. If you see that we can improve it, please open a JIRA and we can rethink those functions.